joern-cli/frontends/swiftsrc2cpg/src/main/scala/io/joern/swiftsrc2cpg/astcreation/AstForExprSyntaxCreator.scala

```diff
@@ -2,38 +2,101 @@ package io.joern.swiftsrc2cpg.astcreation
 import io.joern.x2cpg.frontendspecific.swiftsrc2cpg.Defines
-import io.shiftleft.codepropertygraph.generated.ControlStructureTypes
-import io.shiftleft.codepropertygraph.generated.DispatchTypes
-import io.shiftleft.codepropertygraph.generated.Operators
-import io.shiftleft.codepropertygraph.generated.nodes.NewCall
-import io.shiftleft.codepropertygraph.generated.nodes.NewNode
-import io.shiftleft.codepropertygraph.generated.EvaluationStrategies
+import io.joern.x2cpg.{Ast, ValidationMode}
+import io.shiftleft.codepropertygraph.generated.*
+import io.shiftleft.codepropertygraph.generated.nodes.{NewCall, NewNode}
 
 import scala.annotation.unused
 
 trait AstForExprSyntaxCreator(implicit withSchemaValidation: ValidationMode) {
   this: AstCreator =>
 
-  private def astForListLikeExpr(node: SwiftNode, elements: Seq[SwiftNode]): Ast = {
+  private val MaxInitializers = 1000
+
+  private def astForEmptyListLikeExpr(node: SwiftNode): Ast = {
     val op           = Operators.arrayInitializer
     val initCallNode = callNode(node, code(node), op, DispatchTypes.STATIC_DISPATCH)
+    callAst(initCallNode, List.empty)
+  }
 
-    val MAX_INITIALIZERS = 1000
-    val clauses          = elements.slice(0, MAX_INITIALIZERS)
+  private def astForListLikeExpr(node: SwiftNode, elements: Seq[SwiftNode]): Ast = {
+    if (elements.isEmpty) { astForEmptyListLikeExpr(node) }
+    else {
+      node match {
+        case _: (ArrayExprSyntax | TupleExprSyntax) =>
+          val op           = Operators.arrayInitializer
+          val initCallNode = callNode(node, code(node), op, DispatchTypes.STATIC_DISPATCH)
+
+          val clauses = elements.slice(0, MaxInitializers)
+
+          val args = clauses.map(x => astForNodeWithFunctionReference(x))
+
+          val ast = callAst(initCallNode, args)
+          if (elements.sizeIs > MaxInitializers) {
+            val placeholder =
+              literalNode(node, "<too-many-initializers>", Defines.Any).argumentIndex(MaxInitializers)
+            ast.withChild(Ast(placeholder)).withArgEdge(initCallNode, placeholder)
+          } else {
+            ast
+          }
+        case other =>
+          val blockNode_ = blockNode(node, code(node), Defines.Any)
+
+          scope.pushNewBlockScope(blockNode_)
+          localAstParentStack.push(blockNode_)
+
+          val tmpName      = scopeLocalUniqueName("tmp")
+          val localTmpNode = localNode(node, tmpName, tmpName, Defines.Any).order(0)
+          diffGraph.addEdge(localAstParentStack.head, localTmpNode, EdgeTypes.AST)
+
+          val slicedElements = elements.slice(0, MaxInitializers).toList
+
+          val propertiesAsts = slicedElements.map {
+            case dictElement: DictionaryElementSyntax =>
+              val (lhsNode, rhsAst) = {
+                val key = dictElement.key
+                val fieldName = key match {
+                  case decl: (DeclReferenceExprSyntax | StringLiteralExprSyntax) =>
+                    stripQuotes(code(decl))
```

Comment:
just use an index access here instead of a field access - that allows arbitrary expressions as index, and we don't need such hacks in order to satisfy the constraints of a `FieldIdentifier`